### ISSUE TITLE
websocket & socket connection.params get overwritten by calls that come after the initial call 

### ISSUE THREAD

**Issue by SpeedyC86** [evantahler/actionhero#651]
```
websocket.js:
for(var v in data.params){
connection.params[v] = data.params[v];
}
connection.error = null;
connection.response = {};
server.processAction(connection);
```
If there are two simultaneous calls, connection.params from the first call will be overwritten with the params from the second call and process action will process both actions with the wrong params for the first action.

**Comment by evantahler** [comment]
Have you actually tested this?  This doesn't happen because of the way the actionProcessor (`processAction`) handles state safety.  In older versions of actionhero, we managed this by duplicating the connection object itself, but now we achieve this safety in a simpler/lighter way by simply duplicating the params before processing: [[ https://github.com/evantahler/actionhero/blob/master/initializers/actionProcessor.js#L18 ]] and [[ https://github.com/evantahler/actionhero/blob/master/initializers/actionProcessor.js#L100-L106 ]] 

node/js is single threaded, so only 1 action can be "kicked off" at a time. `server.processAction(connection)` is a synchronous call, so it will continue to block execution until the first async callback is reached, which isn't until after `data.params` is set within the action processor. 

While we test param collisions implicitly in many places in the test suite, I've added another test that calls this out explicitly:  https://github.com/evantahler/actionhero/commit/12d21f520d101609d37147102586f87d18d5b050
